### ISSUE TITLE
Add filter for `Best Trial` on trails list page

### ISSUE THREAD

**Issue by Koen1999** [optuna/optuna-dashboard#654]
# Feature Request
Currently, there is no way to filter for trials with the `Best Trial` attribute on the trials list page of the dashboard. (http://localhost:8080/dashboard/studies/1/trials)
I am aware an overview of best trials is provided on the homepage of the study, but this overview does not contain all details such as duration and notes.

The filter can be implemented as a checkbox in the following menu:
![image](https://github.com/optuna/optuna-dashboard/assets/32361020/319babc4-f73d-44e6-abd9-1def17b0c896)

**Comment by turbotimon** [comment]
I have also missed this from time to time. Just thinking aloud, a ‘sort by value’ could solve this problem and others too?

**Comment by Koen1999** [comment]
@VihaanMotwani , Great to hear you started implementing this! I think there can be multiple best trials simultaneously if several trials obtain the same score. So maybe, it's nice if it also show the number of trials that will be displayed.

I cannot assign the issue to you since I am not a maintainer of this project.

**Comment by VihaanMotwani** [comment]
thanks for the feedback! i wanted to clarify something:

currently, the other filter counts (Complete, Pruned, Fail, etc.) always show the total number of trials in each state, regardless of any filters that are applied. given that, should the “Best Trial (N)” label:

Option A: also display the total number of best trials in the study (matching existing filter behavior)
OR
Option B: display the number of best trials after applying the current filters

im happy to adapt either way!

**Comment by VihaanMotwani** [comment]
yeah that makes sense. i'll make the necessary changes and create a PR shortly!
